File: CarStack.java
/*
* The CarStack Class.
*
* @author  Andi Cucka
* @version 1.0
* @since   2024-03-27
*/

import java.util.ArrayList;
import java.util.List;

/**
* The CarStack() class.
*/
public class CarStack {
    /**
    * Declares a private instance variable.
    */
    private List<String> strStack = new ArrayList<>();

    /**
    * Checks if stack is empty.
    *
    * @return true if the stack is empty, false otherwise
    */
    public boolean isEmpty() {
        return strStack.isEmpty();
    }

    /**
     * Returns top item of stack.
     *
     * @return the peaked string
     */
    public String peak() {
        final String peakedItem;
        if (!isEmpty()) {
            peakedItem = strStack.get(strStack.size() - 1);
        } else {
            peakedItem = null;
        }
        return peakedItem;
    }

    /**
    * Returns stack size.
    *
    * @return the size of the stack
    */
    public int size() {
        return strStack.size();
    }

    /**
    * The showStack() method.
    *
    * @return the values as a string
    */
    public String showStack() {
        String values = "";
        for (int counter = 0; counter < strStack.size(); counter++) {
            values += strStack.get(counter);
            if (counter < strStack.size() - 1) {
                values += ", ";
            }
        }
        return values;
    }

    /**
     * Clears the entire stack.
     */
    public void clear() {
        strStack.clear();
    }

    /**
    * The push() method.
    *
    * @param input from Main.java
    */
    public void push(String input) {
        strStack.add(input);
    }

    /**
    * This method pops a string from the stack.
    *
    * @return the popped string
    */
    public String popItem() {
        final String poppedItem;
        if (!strStack.isEmpty()) {
            poppedItem = strStack.remove(strStack.size() - 1);
        } else {
            poppedItem = null;
        }
        return poppedItem;
    }
}

File: CarStack.ts
/**
 * The CarStack Class.
 *
 * author  Andi Cucka
 * version 1.0
 * since   2024-03-26
 */

export class CarStack {

  /*
  * The constructor for the CarStack string stack class.
  */
  constructor() {
    this.strStack = []
  }

  /*
  * checks if stack is empty.
  */
  public get isEmpty() {
    return this.strStack.length == 0
  }

  /*
  * return top item of stack
  */
  public get peak():string {
    let peakedItem = ""
    if (this.strStack > 0) {
      peakedItem = this.strStack[this.strStack.length - 1]
    } else {
      peakedItem = "nothing to peak"
    }
    return peakedItem
  }
  /*
  * returns stack size
  */
  public get size() {
    return this.strStack.length
  }

  /*
  * returns stack size
  */
  public get size() {
    return this.strStack.length
  }
  
  /*
  * returns the stack as a string
  */
  public get showStack() {
    let values = ""
    for (let counter = 0; counter < this.strStack.length; counter++) {
      values = values + this.strStack[counter] + ", "
    }
    values = values.substring(0, values.length-2)
    return values
  }

  /*
  * clears the entire stack
  */
  public clear() {
    this.strStack = []
  }

  /*
  * pushes a string onto the stack
  */
  public push(input) {
    this.strStack.push(input)
  }

  /*
  * pops a string from the stack
  */
  public popItem() {
    let removed = ""
    // checks if there's at least 1 element in the stack
    if (this.strStack.length > 0) {
      removed = this.strStack[this.strStack.length - 1]
    } else {
      removed = "nothing to remove";
    }
    return removed;
  }
}
